src/Portable.Xaml/Portable.Xaml.Schema/XamlTypeInvoker.cs

```diff
@@ -105,7 +105,15 @@ public virtual void AddToCollection (object instance, object item)
 				}
 
 				addDelegate = CreateAddDelegate(instance, item, collectionType, itemType, key, mode);
-				addDelegate(instance, item);
+
+				try
+				{
+					addDelegate(instance, item);
+				}
+				catch (TargetInvocationException e)
+				{
+					throw e.InnerException;
```

Comment:
Hm, this would actually trash the original stack trace (where the crash actually happened).  I think we should probably leave this try/catch out, unless you have a specific reason for this?

Reply:
I think the purpose of this is throw the original exception raised inside delegate. I  don't see any reason why we can't unwrap delegate invocation from try block.

Reply:
Ohh... I see why it wrapped and re-throwed InnerException after remove this I have failed test: AddToCollectionTypeMismatch(). ```Assert.Throws<ArgumentException> (() => i.AddToCollection (l, "5"));``` We must leave this or we can rewrite test and scan ```ArgumentException``` in ``InnerException``` or something else? @cwensley what do you think about this?

Reply:
@cm4ker take a look at [ExceptionDispatchInfo](https://stackoverflow.com/a/17091351/981187), it can be used to rethrow the inner exception without losing the stack trace.  I've used it before [here](https://github.com/picoe/Eto/blob/develop/test/Eto.Test/UnitTests/TestBase.cs#L131) and it works well.